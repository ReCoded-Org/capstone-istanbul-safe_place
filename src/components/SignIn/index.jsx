import React from "react";
import { Container, Row, Col } from "react-bootstrap";
import { useHistory } from "react-router-dom";
import SignInForm from "../SignInForm";
import womenGroupHuggingTogether from "../../images/womenGroupHuggingTogether.png";
import googleIcon from "../../images/icons/googleIcon.svg";
import twitterIcon from "../../images/icons/twitterIcon.svg";
import facebookIcon from "../../images/icons/facebookIcon.svg";
import { setFirebaseProvider } from "../../utils/setFirebaseProvider";
import firebase from "../../firebaseConfig";

import "./index.scss";

export default function SignIn() {
  const history = useHistory();

  const signUpFailed = (error) => {
    alert(error);
  };

  const handleSignUpWithProvider = React.useCallback(
    async (providerName) => {
      const firebaseProvider = setFirebaseProvider(providerName);

      firebase
        .auth()
        .signInWithPopup(firebaseProvider)
        .then(function (result) {
          history.push("/");
        })
        .catch(function (error) {
          // Handle errors here.
          signUpFailed(error);
        });
    },
    [history]
  );

  return (
    <Container fluid="md" className="signInSection">
      <Row>
        <Col className="signInImg" md={6} xs={12}>
          <img
            src={womenGroupHuggingTogether}
            alt="Women Group Hugging Together"
          />
        </Col>
        <Col className="signInForm" md={6} xs={12}>
          <h2>Welcome Back!</h2>
          <h4>Sign in with</h4>
          <ul className="signInIcons">
            <li className="signInIcon">
              <a href="#/" onClick={() => handleSignUpWithProvider("twitter")}>
                <img src={twitterIcon} alt="Twitter icon" />
              </a>
            </li>
            <li className="signInIcon">
              <a href="#/" onClick={() => handleSignUpWithProvider("facebook")}>
                <img src={facebookIcon} alt="Facebook icon" />
              </a>
            </li>
            <li className="signInIcon">
              <a href="#/" onClick={() => handleSignUpWithProvider("google")}>
                <img src={googleIcon} alt="Google icon" />
              </a>
            </li>
          </ul>
          <hr className="divider" />

          <SignInForm
            submit={() => {
              /* TODO: implement the signing in functionality */
            }}
          />

          <p>
            New here?
            <a href="/sign-up"> Create an account</a>
          </p>
        </Col>
      </Row>
    </Container>
  );
}
