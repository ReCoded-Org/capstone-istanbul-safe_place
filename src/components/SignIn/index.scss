@import "../../style/variables";
@import "../../style/buttons";
@import "../../style/headings";
@import "../../style/bootstrapImports";

.singInSection {
  /* All section style */
  width: 100%;

  /* Image Style */
  .signInImg {
    background-color: $secondaryColor;
  }

  /* Form Style */
  .signInForm {
    background-color: $lightColor;
  }

  /* Heading Style */
  h2 {
    @include mainHeading;
    color: $primaryColor;
    margin-bottom: 2rem;
  }

  h4 {
    font-size: 1.25rem;
    color: $primaryColor;
    margin-bottom: 1rem;

    &::before {
      display: inline-block;
      border-top: 0.4rem solid #fe8f7c;
      border-radius: 10px;
      width: 56px;
      height: 0px;
      content: "";
      margin: auto 0.5rem auto 0;
    }
  }

  /* Link Style */
  a {
    color: $primaryColor;

    &:hover {
      text-decoration: none;
      color: $warningColor;
    }
  }

  /* Section devider Style */
  hr.divider {
    overflow: visible;
    padding: 0;
    margin-bottom: 10px;
    border: none;
    border-top: 1px solid #e0e0e0;
    color: #6e6d7a;
    text-align: center;

    &::after {
      content: "Or";
      display: inline-block;
      position: relative;
      top: -13px;
      font: normal 14px/14px "Haas Grot Text R Web", "Helvetica Neue", Helvetica,
        Arial, sans-serif;
      padding: 0 16px;
      background: $lightColor;
    }
  }

  /* Icons Style */
  .signInIcons {
    box-sizing: border-box;
    display: flex;
    justify-content: space-around;
  }
  .signInIcon {
    list-style: none;
    background-color: $secondaryColor;
    border-radius: 50px;
    padding: 0.6rem;
    margin: 1rem 0;

    &:hover {
      background-color: $primaryColor;
    }
  }

  .signInIcon img {
    width: 2.5rem;
    height: 2.5rem;
  }
}
