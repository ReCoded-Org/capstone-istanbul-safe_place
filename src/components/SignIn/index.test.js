import React from "react";
import renderer from "react-test-renderer";
import SignIn from "./index";

it("renders correctly", () => {
  const tree = renderer.create(<SignIn />).toJSON();
  expect(tree).toMatchSnapshot();
});
