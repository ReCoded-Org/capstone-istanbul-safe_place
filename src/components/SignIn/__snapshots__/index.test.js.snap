// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="singInSection"
>
  <div
    className="contactContainer container-md"
  >
    <div
      className="row"
    >
      <div
        className="signInImg col-md-6 col-12"
      >
        <img
          alt="Women Group Hugging Together"
          src="womenGroupHuggingTogether.png"
        />
      </div>
      <div
        className="signInForm col-md-6 col-12"
      >
        <h2>
          Welcome Back!
        </h2>
        <h4>
          Sign in with
        </h4>
        <ul
          className="signInIcons"
        >
          <li
            className="signInIcon"
          >
            <a
              href="/"
            >
              <img
                alt="Twitter icon"
                src="twitterIcon.svg"
              />
            </a>
          </li>
          <li
            className="signInIcon"
          >
            <a
              href="/"
            >
              <img
                alt="Facebook icon"
                src="facebookIcon.svg"
              />
            </a>
          </li>
          <li
            className="signInIcon"
          >
            <a
              href="/"
            >
              <img
                alt="Google icon"
                src="googleIcon.svg"
              />
            </a>
          </li>
        </ul>
        <hr
          className="divider"
        />
        <div>
          sign in form
        </div>
        <p>
          New here?
          <a
            href="/sign-up"
          >
            Create an account
          </a>
        </p>
      </div>
    </div>
  </div>
</section>
`;
